File: java/1669_MergeInBetweenLinkedLists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        int i = 1;
        ListNode curr = list1;
        while (i < a) {
            curr = curr.next;
            i++;
        }
        
        ListNode temp = curr.next;
        ListNode head = list1;
        if (a > 0) curr.next = list2;
        else head = list2;
        
        while (i < b) {
            temp = temp.next;
            i++;
        }
        
        ListNode second = list2;
        while (second.next != null) {
            second = second.next;
        }
        
        second.next = temp.next;
        return head;
    }
}

File: python/1669_MergeInBetweenLinkedLists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        i = 1
        curr = list1
        while i < a:
            i += 1
            curr = curr.next
            
        temp = curr.next
        curr.next = list2
        head = list1 if a > 0 else list2
        
        while i < b:
            i += 1
            temp = temp.next
        
        second = list2
        while second.next:
            second = second.next
            
        second.next = temp.next
        return head
        
